core/drivers/riscv_zkr_rng.c

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2024 Andes Technology Corporation
+ */
+
+#include <crypto/crypto.h>
+#include <kernel/panic.h>
+#include <riscv.h>
+#include <rng_support.h>
+#include <tee/tee_cryp_utl.h>
+
+#define OPST_BIST 0b00
+#define OPST_WAIT 0b01
+#define OPST_ES16 0b10
+#define OPST_DEAD 0b11
+
+static unsigned long read_seed(void)
+{
+	unsigned long ret = 0;
+
+	/*
+	 * The seed register must be accessed using CSR read-write
+	 * instructions. The write operation is ignored and serves
+	 * to indicate polling and flushing.
+	 */
+	asm volatile("csrrw %0, seed, x0" : "=r" (ret) : : "memory");
```

Comment:
In the OP-TEE project, we avoid mixing inline assembly with C code. Implementing a single instruction like this as a static inline function in `core/arch/riscv/include/riscv.h` is preferred.